src/Integration/LocalServices/RuleSetInspector.cs

```diff
@@ -0,0 +1,144 @@
+            //  => either way won't weaken.
+            // 
+            // (●) By default i.e. the way we generate the targetRuleSet (which we expect it to be the project level ruleset),
+            // the includes cannot weaken the base line ruleset (which we expect it to be the solution level ruleset)
+            // due to to the include merge rules which prefer strictness and the fact that we use Default as the Include Action.
+            // 
+            // (●) The only thing that can weaken the rules in practice, in the default case, are ruleset level rules 
+            // i.e what you get when you modify the ruleset by using the ruleset editor UI.
+            // 
+            // (●) In the none default case, when the user decides to manually edit the file, setting the baseline include
+            // to None or removing it will use the other Includes' Action settings which could weaken the target ruleset.
+            // 
+            // With that in mind, we will find "conflicts" of two types - rule with weaker Action or a missing rule.
+            // (assuming that all the rules were found with the help of ruleSetDirectories).
+            // 
+            // In terms of how to fix the conflict...  we could either just add all the conflicted rules into the project. That 
+            // will not be optimal in terms of the user or the rebind/update experience. A better approach would be something like:
+            // 1. Reset the baseline Include to Action=Default
+            // 2. If there are still conflicts, remove all the conflicting rules which are directly on target ruleset.
+            // 3. If there are still conflicts, add the remaining conflicting rules directly under the target with Action=Warning
+
+            var effectiveRulesMap = target.GetEffectiveRules(ruleSetDirectories, providers, this.EffectiveRulesErrorHandler)
+                .ToDictionary(r => r.FullId, StringComparer.OrdinalIgnoreCase);
+
+            var deprioritizedRules = baseLine.Rules.Select(r =>
+            {
+                RuleReference reference;
+                if (effectiveRulesMap.TryGetValue(r.FullId, out reference))
+                {
+                    Debug.Assert(reference.Action != RuleAction.None, "Expected to be found in the missing set");
+
+                    if (reference.Action == RuleAction.Info)
+                    {
+                        return reference;
+                    }
+                }
+
+                return null;
+            }).Where(r => r != null);
+
+            var disabledRules = baseLine.Rules.Where(r => !effectiveRulesMap.ContainsKey(r.FullId));
+
+            return new RuleConflictInfo(disabledRules, deprioritizedRules);
+        }
+
+        private void EffectiveRulesErrorHandler(string message, Exception error)
+        {
+            VsShellUtils.WriteToGeneralOutputPane(this.serviceProvider, Strings.UnexpectedRuleSetInpectorError, nameof(RuleSetInspector), message);
```

Comment:
The class name "RuleSetInspector" isn't going to mean much to the end user. There's nothing to indicate that the message relates to the SonarQube vsix.